server/controllers/docHandlers.js

```diff
@@ -51,6 +51,50 @@ const retrieveDoc = (req, res) => {
   });
 };
 
+const lockDoc = (req, res) => {
+  if(!req.isAuthenticated()){
+    res.status(401).json({error:"Must be logged in to perform this action"});
+  }
+  else{
+    Doc.findOne({_id:req.params.documentId}, function(lockErr, lockResult){
+      if(lockErr){
+        return res.status(500).json({error:"Failed to check the lock on document"});
+      }
+      if(lockresult.locked) {
+        return res.status(403).json({error:"Another user is currently making changes to the document"});
+      }
+      Doc.updateOne({_id:req.params.documentId}, {locked: req.user._id}, function(err, result){
```

Comment:
Making another query is not good, you can update it without making another query